compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -775,6 +775,35 @@ declare_lint! {
     "detects macros that were not used"
 }
 
+declare_lint! {
+    /// The `unused_macro_rules` lint detects macro rules that were not used.
```

Comment:
I'd add some wording saying that the rules in question are inside the macro, and give an explicit example.
Otherwise it's confusable with whole `macro_rules` items.